src/Controller/Controller.php

```diff
@@ -201,9 +201,16 @@ public function __construct(
         }
 
         $this->modelFactory('Table', [$this->getTableLocator(), 'get']);
-        $plugin = $this->request->getParam('plugin');
-        $modelClass = ($plugin ? $plugin . '.' : '') . $this->name;
-        $this->_setModelClass($modelClass);
+
+        if ($this->modelClass === null) {
```

Comment:
Perhaps `$modelClass` should always be overwritten if `$defaultTable` is not null, since we eventually want to deprecate  using of `loadModel()` and `$modelClass`.

```
if ($this->defaultTable !== null) {
    $this->modelClass = $this->defaultTable;
}
```

Reply:
> Perhaps $modelClass should always be overwritten if $defaultTable is not null, since we eventually want to deprecate using of loadModel() and $modelClass.

Agreed. The new system should replace the old one.